graphene/types/schema.py

```diff
@@ -357,7 +371,13 @@ def create_fields_for_type(self, graphene_type, is_input_type=False):
                     deprecation_reason=field.deprecation_reason,
                     description=field.description,
                 )
-            field_name = field.name or self.get_name(name)
+            if field.name:
+                field_name = field.name
+            else:
+                if graphene_type._meta.name in self.root_type_names:
+                    field_name = self.add_prefix_to_field_name(name)
+                else:
+                    field_name = self.get_name(name)
```

Comment:
I'm not completely sure but I think that even if the name comes from `field.name` we want to add the prefix... would be good to add a unit test for this.

```suggestion
            field_name = field.name or self.get_name(name)
            if graphene_type._meta.name in self.root_type_names:
                field_name = self.add_prefix_to_field_name(name)
```

Reply:
Looking again at the code I see that we bypass `auto_camelcase` when the field name is explicitly defined, shouldn't we expect the same for `type_name_prefix` ? 

I'm not sure neither to be honest

Reply:
yeah I saw that we bypass the auto camel case but I am not sure it makes sense to remove the prefixing... 🤷‍♂️